gui/default/syncthing/core/syncthingController.js

```diff
@@ -12,14 +12,30 @@ angular.module('syncthing.core')
         var online = false;
         var restarting = false;
 
+        // window.metadata is set in /meta.js which is behind API auth,
+        // so this will be present only when authenticated.
+        var authenticated = Boolean(window.metadata);
```

Comment:
What do you mean by that - there's no "logout"? This is locally scoped to the controller, so it gets recalculated e.g. on a refresh.

Reply:
Exactly, it is not clear to me how one can reach a state again where a new login is allowed. Maybe we don't need it now because there is only one user, but I think some people will search for a way to log out before leaving the GUI.

The code here just looks suspicious to me because it's not clear how that metadata variable somehow proves that authentication had happened and since when, i.e. does it ever disappear from the `window` again?

Reply:
I've added a logout feature in a meta-PR that would merge into this one: https://github.com/emlun/syncthing/pull/2 . Please review!

Does the comment adequately explain how the variable "proves" authentication?

Reply:
The comment is fine. It just leaves the question what exactly is proven and how long that proof is being cached for other code to possibly rely on. For example I don't know if the browser may cache `meta.json` and therefore come to the wrong conclusion here because the cached code is executed and sets the variable, without actually asking the backend.

Reply:
Hm, good point about caching - there is a `noCacheMiddleware` that's applied to the `/rest/` path of the backend API, but `/meta.js` is not covered by that. It probably should also be, or we should instead use some other path under `/rest/` to check if we're authenticated.

Reply:
I made this a bit more explicit in commit 947852fdacfdac18734c022d5a8fc43b5e6e4978: `/meta.js` now sets an `"authenticated"` variable, and is now also wrapped with a `noCacheMiddleware` to prevent cache issues. Does that look okay?

Reply:
Looks good, though I'd have kept the comment for clarity.

Reply:
Ok, I've re-added a comment to the same effect.